ethereumj-core/src/main/java/org/ethereum/cli/CLIInterface.java

```diff
@@ -99,10 +81,81 @@ public static void call(String[] args) {
         }
     }
 
-    private static Boolean interpret(String arg) {
+    // show help
+    private static void processHelp(String arg) {
+        if ("--help".equals(arg)) {
+            printHelp();
+
+            System.exit(1);
+        }
+    }
+
+    private static boolean processConnectOnly(String arg, Map<String, Object> cliOptions) {
+        if ("-connectOnly".equals(arg))
+            return false;
+
+        cliOptions.put(SystemProperties.PROPERTY_PEER_DISCOVERY_ENABLED, false);
+
+        return true;
+    }
+
+    // override the db directory
+    private static boolean processDbDirectory(String arg, String db, Map<String, Object> cliOptions) {
+        if (!"-db".equals(arg))
+            return false;
+
+        logger.info("DB directory set to [{}]", db);
+
+        cliOptions.put(SystemProperties.PROPERTY_DB_DIR, db);
 
-        if (arg.equals("on") || arg.equals("true") || arg.equals("yes")) return true;
-        if (arg.equals("off") || arg.equals("false") || arg.equals("no")) return false;
+        return true;
+    }
+
+    // override the listen port directory
+    private static boolean processListenPort(String arg, String port, Map<String, Object> cliOptions) {
+        if (!"-listen".equals(arg))
+            return false;
+
+        logger.info("Listen port set to [{}]", port);
+
+        cliOptions.put(SystemProperties.PROPERTY_LISTEN_PORT, port);
+
+        return true;
+    }
+
+    // override the connect host:port directory
+    private static boolean processConnect(String arg, String connectStr, Map<String, Object> cliOptions) throws URISyntaxException {
+        if (!arg.startsWith("-connect"))
+            return false;
+
+        logger.info("Connect URI set to [{}]", connectStr);
+        URI uri = new URI(connectStr);
+
+        if (!"enode".equals(uri.getScheme()))
+            throw new RuntimeException("expecting URL in the format enode://PUBKEY@HOST:PORT");
+
+        List<Map<String, String>> peerActiveList = Collections.singletonList(Collections.singletonMap("url", connectStr));
+
+        cliOptions.put(SystemProperties.PROPERTY_PEER_ACTIVE, peerActiveList);
+
+        return true;
+    }
+
+    // process database reset
+    private static boolean processDbReset(String arg, String reset, Map<String, Object> cliOptions) {
+        if (!"-reset".equals(arg))
+            return false;
+
+        Boolean resetFlag = interpret(reset);
+        logger.info("Resetting db set to [{}]", resetFlag);
+        cliOptions.put(SystemProperties.PROPERTY_DB_RESET, resetFlag.toString());
+
+        return true;
+    }
+
+    private static Boolean interpret(String arg) {
+        if ("on".equals(arg) || "true".equals(arg) || "yes".equals(arg)) return true;
```

Comment:
Could you replace both lines in favor of BooleanUtils.toBooleanObject(String) with throwing error when null returned. I guess it would be safer.

Reply:
Yes! Done

Reply:
Add something like
```
if (resetFlag == null) {
    throw new Error("Can't interpret the answer: " + arg);
}
```
You've lost this case

Reply:
Sure, good catch... done